R/layout.R

```diff
@@ -0,0 +1,49 @@
+#' Modify the layout of a plotly visualization
+#' 
+#' @param p A plotly object.
+#' @param ... Arguments to the layout object. For documentation,
+#' see \url{https://plot.ly/r/reference/#Layout_and_layout_style_objects}
+#' @param data A data frame to associate with this layout (optional). If not 
+#' provided, arguments are evaluated using the data frame in \code{\link{plot_ly}()}.
+#' @author Carson Sievert
+#' @export
+layout <- function(p, ..., data = NULL) {
+  UseMethod("layout")
+}
+
+#' @export
+layout.matrix <- function(p, ..., data = NULL) {
+  # workaround for the popular graphics::layout() function
+  # https://github.com/ropensci/plotly/issues/464
+  graphics::layout(p, ...)
+}
+
+#' @export
+layout.plotly <- function(p, ..., data = NULL) {
+  p <- add_data(p, data)
+  attrs <- list(...)
+  if (is.null(p$x$cur_data)) {
+    nlayouts <- length(p$x$layout)
+    p$x$layout <- c(p$x$layout, attrs)
+  } else {
+    p$x$layoutAttrs[[p$x$cur_data]] <- attrs
```

Comment:
This will overwrite any previous defined `layout`.  Is this intentional?

```
data(economics, package = "ggplot2")
# basic time-series plot
p <- plot_ly(economics, x = ~date, y = ~uempmed)
# add a loess smoother
p2 <- add_trace(p, y = ~fitted(loess(uempmed ~ as.numeric(date))))
# add a title
p3 <- layout(p2, title = "Median duration of unemployment (in weeks)")
# change the font
layout(p3, font = list(family = "Courier New, monospace"))
```

The `title` is lost.

![image](https://cloud.githubusercontent.com/assets/837910/16155788/9acf41f8-3476-11e6-83ea-0c8f2f01a707.png)